test/decl/enum/enumtest.swift

```diff
@@ -333,3 +333,80 @@ enum Lens<T> {
 enum HasVariadic {
   case variadic(x: Int...) // expected-error {{variadic enum cases are not supported}}
 }
+
+// SR-2176
+enum Foo {
+  case bar
+  case none
+}
+
+let _: Foo? = .none // expected-warning {{assuming you mean 'Optional<Foo>.none'; did you mean 'Foo.none' instead?}} {{14-14=Optional}} {{14-14=Foo}}
+let _: Foo?? = .none // expected-warning {{assuming you mean 'Optional<Foo>.none'; did you mean 'Foo.none' instead?}} {{15-15=Optional}} {{15-15=Foo}}
```

Comment:
This warning is incorrect here, unless I'm mistaken. The compiler assumes that this is `Optional<Optional<Foo>>.none`:

```swift
enum X { case none }
var x: X?? = .none
if x == Optional<Optional<X>>.none {
  print("x == Optional<Optional<X>>.none")
}
// Prints "x == Optional<Optional<X>>.none"
```

Reply:
Oh yeah, my bad. I hardcoded `Optional` in the diagnostic message and used the unwrapped type in the `<>`. Fixed it now by using the canonical type.